File: src/main/java/github/cweijan/ultimate/util/json/deserializer/LocalDateTimeDeserializer.java
package github.cweijan.ultimate.util.json.deserializer;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonDeserializer;
import github.cweijan.ultimate.util.DateUtils;

import java.io.IOException;
import java.time.LocalDateTime;

/**
 * @author cweijan
 * @version 2019/9/5 11:34
 */
public class LocalDateTimeDeserializer extends JsonDeserializer<LocalDateTime> {
    @Override
    public LocalDateTime deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException, JsonProcessingException {
        return (LocalDateTime) DateUtils.toDateObject(jsonParser.getText().trim(), LocalDateTime.class);
    }
}

File: src/main/java/github/cweijan/ultimate/json/deserializer/LocalDateTimeDeserializer.kt
package github.cweijan.ultimate.json.deserializer

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.JsonDeserializer
import github.cweijan.ultimate.util.DateUtils
import java.time.LocalDateTime

class LocalDateTimeDeserializer : JsonDeserializer<LocalDateTime>() {

    override fun deserialize(p: JsonParser, ctxt: DeserializationContext): LocalDateTime? {

        return DateUtils.toDateObject(p.text.trim(), LocalDateTime::class.java) as LocalDateTime?
    }
}
